lib/validate.js

```diff
@@ -13,7 +13,24 @@ function validate (state, doc) {
     return state.validate(doc)
   })
   .catch(function (error) {
-    error.name = 'ValidationError'
-    throw error
+    var err = new Error()
```

Comment:
what do you think about renaming the passed argument from `error` to `rejectValue` and `var err` to `var error`? I try to avoid abbreviations as much as possible as a general rule

Reply:
Isn't it misleading to make the argument name `rejectValue`, considering it can be an error throw directly as well? Although, I can't think of any other name better suited to it :confused:

Reply:
technically a thrown error is captured by a promise and turned into the value that the promise rejects with :)

Reply:
Dayum. This task has been the best on learning how Promises work :stuck_out_tongue: